feat(BuildCommand.php): add lifecycle hooks for prehtml and prepdf

bmcminn opened the pull request:
Adds user editable functions to ibis.php that allows users to modify
content via global event hook callbacks.

Fixes issue #13

themsaid commented:
I think it's better if you don't have to define global functions. Just include a closure in the config file to be invoked within the build command. Also can you please include an example of how you're using both callbacks in a real scenario?

bmcminn commented:
Yeah it's not ideal. 

Reason I went this route was because the method signature for `BuildCommand.php@buildHtml()` doesn't have an argument for `$config`.

Would it make more sense to make `$config` a property of the `BuildCommand` class so we can just reference that in places as opposed to passing the `$config` value around from method to method?

bmcminn commented:
I think that's the final version of this feature.

To recap, I moved the `$config` variable from being a local method argument to being a private class field of `BuildCommand` class. I also updated each local `$config` reference to `$this->_config` and removed all `$config` arguments from the method signatures to avoid ambiguity.

So far I'm pretty happy with how this turned out.

stefanbauer commented on src/Commands/BuildCommand.php:
```diff
@@ -33,6 +33,12 @@ class BuildCommand extends Command
+     * User ibis.php config settings
+     * @var array
+     */
+    private $_config;
```
This `_` is weird. The times when underscores were used to mark variables as private are long gone. We should go with simple `$config` if we follow that approach.

bmcminn replied:
Cleaned this one up in latest push.

stefanbauer commented on src/Commands/BuildCommand.php:
```diff
@@ -62,7 +68,7 @@ public function execute(InputInterface $input, OutputInterface $output)
 
         $currentPath = getcwd();
-        $config = require $currentPath.'/ibis.php';
+        $this->_config = require $currentPath.'/ibis.php';
```
Maybe it's better to load that configuration in the constructor?

bmcminn replied:
I created a `__construct()` method that calls `super::__construct()` before assigning `$this->config`.

This change also required the `$currentPath` value in the constructor, so I moved it to another private field in the class and cleaned up all the method signatures and updated the all references to use `$this->currentPath` instead.

bmcminn commented on src/Commands/BuildCommand.php:
```diff
@@ -107,6 +126,10 @@ protected function buildHtml(string $path)
     {
         $this->output->writeln('<fg=yellow>==></> Parsing Markdown ...');
 
+        if (is_callable($this->config['prehtml'] ?? null)) {
```
Invoking this output call here so it only renders once in the console as opposed to per file.

Might be nice to render this message per file along with the active filename so users can see which file might be breaking for them?

bmcminn commented:
I think this feature is in pretty good shape for final review.

bmcminn commented:
@themsaid any thoughts on accepting this PR?

bmcminn commented:
@themsaid *bump
